### PR TITLE
Update to Osctrl docker-compose and update to Docker CI/CD

### PR DESCRIPTION
**Pull request by CptOfEvilMinions**
*Changes*
* Moved PROD [docker-compose](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/docker-compose.yml) has been moved to the root of the repo
* Added PROD docker-compose example `.env` file - https://github.com/jmpsec/osctrl/blob/cc558ecbe921e4d0f57bd443c559e9d957892fa1/.env.example
* Updated Dockerfile for [osctrl-admin](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/cicd/Dockerfile-osctrl-admin)
* Updated Dockerfile for [osctrl-api](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/cicd/Dockerfile-osctrl-api)
* Updated Dockerfile for [osctrl-cli](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/cicd/Dockerfile-osctrl-cli)
* Updated Dockerfile for [osctrl-tls](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/cicd/Dockerfile-osctrl-tls)
* Added Dockerfile for a [test Osquery agent ](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/docker/Dockerfile-osquery)
* Update docker-compose [README](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/docker/README.md)
* [Update NGINX config for Osctrl for Docker](https://github.com/jmpsec/osctrl/blob/8a482f48e2c37c09f7868c8d98730d78c85fdec3/deploy/docker/conf/nginx/osctrl.conf)
* Create a Docker [entrypoint script](https://github.com/jmpsec/osctrl/pull/179/files#diff-b98bbf825bc7a878270dd08c20ba7ec6cd63787d0a39a9aa8d675908d5187f57) for osctrl-cli to create an osctrl env using ENV vars

### CODE REVIEW

File: deploy/docker/docker-compose-prod.yml [hunk 1]
```diff
@@ -34,12 +34,25 @@ services:
...
     container_name: 'osctrl-tls'
     restart: unless-stopped
-    image: osctrl-tls:${OSCTRL_VERSION}
-    build:
-      <<: *build_osctrl
-      target: osctrl-tls
+    user: "0:0"
+    environment:
+      #### TLS settings ####
+      - TLS_SERVER=false
```
Comment (javuto) [inline deploy/docker/docker-compose-prod.yml:44]: `TLS_SERVER` defaults to `false` so you can skip this assignment

**Review by javuto — COMMENTED** [review]
`api/main.go` and `admin/main.go` should not be removed

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    container_name: 'osctrl-api'
+    restart: unless-stopped
+    image: jmpsec/osctrl-api:v${OSCTRL_VERSION}
+    environment:
+      #### TLS settings ####
+      - SERVICE_LISTENER=0.0.0.0
+      - SERVICE_PORT=9002
+      - SERVICE_HOST=0.0.0.0
+      - SERVICE_AUTH=jwt
+      - SERVICE_CONFIG=false
```
Comment (javuto) [inline docker-compose.yml:107]: No need to set this to `false`, that is the default value

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      start_period: 40s
+
+  ######################################### osctrl-admin #########################################
+  osctrl-admin:
+    container_name: 'osctrl-admin'
+    restart: unless-stopped
+    image: jmpsec/osctrl-admin:v${OSCTRL_VERSION}
+    environment:
+      #### TLS settings ####
+      - SERVICE_CONFIG=false
```
Comment (javuto) [inline docker-compose.yml:69]: No need to set this to `false`, that is the default value

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+      - SERVICE_LISTENER=0.0.0.0
+      - SERVICE_PORT=9001
+      - SERVICE_HOST=0.0.0.0
+      - SERVICE_AUTH=db
+      - JWT_SECRET=${JWT_SECRET}
+      - SERVICE_LOGGER=stdout      
+      #### Database settings ####
+      - DB_HOST=postgres
+      - DB_PORT=5432
```
Comment (javuto) [inline docker-compose.yml:78]: This is the default port, we can skip this

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+      #### TLS settings ####
+      - SERVICE_CONFIG=false
+      - SERVICE_LISTENER=0.0.0.0
+      - SERVICE_PORT=9000
+      - SERVICE_HOST=0.0.0.0
+      - SERVICE_AUTH=none
+      - SERVICE_LOGGER=stdout      
+      #### Database settings ####
+      - DB_HOST=postgres
+      - DB_PORT=5432
```
Comment (javuto) [inline docker-compose.yml:44]: This is the default port, we can skip this

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+      - DB_NAME=${POSTGRES_DB_NAME}
+      - DB_USER=${POSTGRES_DB_USERNAME}
+      - DB_PASS=${POSTGRES_DB_PASSWORD}
+      #### Redis settings ####
+      - REDIS_HOST=redis
+      - REDIS_PORT=6379
```
Comment (javuto) [inline docker-compose.yml:50]: This is the default port, we can skip this

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (javuto) [inline docker-compose.yml:84]: This is the default port, we can skip this

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+      - SERVICE_LISTENER=0.0.0.0
+      - SERVICE_PORT=9002
+      - SERVICE_HOST=0.0.0.0
+      - SERVICE_AUTH=jwt
+      - SERVICE_CONFIG=false
+      - JWT_SECRET=${JWT_SECRET}
+      - SERVICE_LOGGER=stdout      
+      #### Database settings ####
+      - DB_HOST=postgres
+      - DB_PORT=5432
```
Comment (javuto) [inline docker-compose.yml:112]: This is the default port, we can skip this

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+      - CERT_FILE=/opt/osctrl/config/osctrl.crt
+      - HOST=osctrl-nginx
+      #### osctrl admin ####
+      - OSCTRL_USER=${OSCTRL_USER}
+      - OSCTRL_PASS=${OSCTRL_PASS}
+      #### Database settings ####
+      - DB_CONFIG=true
+      - DB_JSON=/opt/osctrl/config/db.json
+      - DB_HOST=postgres
+      - DB_PORT=5432
```
Comment (javuto) [inline docker-compose.yml:179]: This is the default port, we can skip this

File: docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+    environment:
+      #### osctrl env ####
+      - ENV_NAME=dev
+      - CERT_FILE=/opt/osctrl/config/osctrl.crt
+      - HOST=osctrl-nginx
+      #### osctrl admin ####
+      - OSCTRL_USER=${OSCTRL_USER}
+      - OSCTRL_PASS=${OSCTRL_PASS}
+      #### Database settings ####
+      - DB_CONFIG=true
```
Comment (javuto) [inline docker-compose.yml:176]: Using a JSON file will overwrite any env variable so if the configuration is via `db.json`, the rest of `DB_` are not needed

File: docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+      dockerfile: deploy/docker/Dockerfile-osquery
+      args:
+        OSQUERY_VERSION: ${OSQUERY_VERSION}
+        OSCTRL_VERSION: ${OSCTRL_VERSION}
+    environment:
+      #### osquery enroll ####
+      - ENV_NAME=dev
+      - HOST=osctrl-nginx
+      #### Database settings ####
+      - DB_CONFIG=true
```
Comment (javuto) [inline docker-compose.yml:209]: If JSON is used, no need for env variables

File: docker-compose.yml (hunk 10 above)
Comment (CptOfEvilMinions) [reply docker-compose.yml:176]: See comment in Slack

File: docker-compose.yml (hunk 11 above)
Comment (CptOfEvilMinions) [reply docker-compose.yml:209]: See comment in Slack

File: deploy/docker/Dockerfile-osctrl-prod [hunk 12]
```diff
@@ -1,228 +0,0 @@
-######################################## osctrl-base ########################################
```
Comment (javuto) [inline deploy/docker/Dockerfile-osctrl-prod:1]: Removing this?

**Review by javuto — APPROVED** [review]

**Pull request merged**